### PR TITLE
Match latest Secureframe GraphQL changes

### LINKED ISSUE #14: graphql: unexpected status code: 504 (will retry)
```
2024/07/08 14:59:34 Getting Secureframe tests for soc2_alpha ...
2024/07/08 14:59:34 POST to https://app.secureframe.com/graphql with "Bearer" token: 2308 bytes
[20](https://github.com/chainguard-dev/internal/actions/runs/9841600735/job/27169890739#step:6:21)24/07/08 15:00:34 unexpected status code: 504 (will retry)
2024/07/08 15:00:36 Secureframe test query failed: get company test v2s: query: unexpected status code: 400
panic: Secureframe test query failed: get company test v2s: query: unexpected status code: 400
goroutine 1 [running]:
log.Panicf({0x80e08c?, 0xb84f00?}, {0xc000157f30?, 0x24?, 0xc00002[21](https://github.com/chainguard-dev/internal/actions/runs/9841600735/job/27169890739#step:6:22)c2?})
	/opt/hostedtoolcache/go/1.22.5/x64/src/log/log.go:439 +0x65
main.main()
	/home/runner/work/internal/internal/secureframe-issue-sync/main.go:61 +0x2c5
```

### PR DESCRIPTION
**Pull request by tstromberg**
Fixes #14

### CODE REVIEW

File: pkg/secureframe/graphql.go [hunk 1]
```diff
@@ -308,13 +345,13 @@ func query(ctx context.Context, token string, in interface{}, out interface{}) e
 		return fmt.Errorf("read: %w", err)
 	}
 
-	// log.Printf("response: %s", rb)
+	log.Printf("response: %s", rb)
```
Comment (egibs) [inline pkg/secureframe/graphql.go:348]: Can these be commented out now that the changes work?

File: pkg/secureframe/graphql.go [hunk 2]
```diff
@@ -572,166 +609,251 @@ func getCompanyTest(ctx context.Context, companyID string, token string, id stri
...
+		if requestCount > maxRequests {
+			return found, fmt.Errorf("made too many requests")
+		}
+
+		ts, meta, err := getCompanyTestV2s(ctx, companyID, token, reportKey, page)
+		if err != nil {
+			return nil, fmt.Errorf("get company test v2s: %w", err)
+		}
+		found = append(found, ts...)
+		//time.Sleep(time.Duration(page) * time.Second)
```
Comment (egibs) [inline pkg/secureframe/graphql.go:639]: Can this be removed with the new rate-limiting method being used?

File: pkg/secureframe/graphql.go [hunk 3]
```diff
@@ -572,166 +609,251 @@ func getCompanyTest(ctx context.Context, companyID string, token string, id stri
...
 	// If we had a properly documented GraphQL API, we could get everything in a single query.
-	nts := []Test{}
-	for x, t := range tests {
+	detailed := []Test{}
+
+	for x, t := range found {
+		// No need for details in these cases
 		if t.Pass || !t.Enabled {
-			nts = append(nts, t)
+			detailed = append(detailed, t)
```
Comment (egibs) [inline pkg/secureframe/graphql.go:653]: The comment makes it seem like only `continue` should be used here; is that correct?

File: pkg/secureframe/graphql.go (hunk 1 above)
Comment (tstromberg) [reply pkg/secureframe/graphql.go:348]: Yes, but I'd prefer to leave them in as any breakage almost certainly means uncommenting these again.

Comment (egibs) [reply pkg/secureframe/graphql.go:348]: Yea, good point 🥲

File: pkg/secureframe/graphql.go (hunk 3 above)
Comment (tstromberg) [reply pkg/secureframe/graphql.go:653]: Good question. Not quite - "t" is the undetailed data struct that we append to the test list. We still need the undetailed data to know if we can close an existing issue.

**Pull request merged**